scs2-session-logger/src/main/java/us/ihmc/scs2/session/tools/RobotModelLoader.java

```diff
@@ -200,13 +198,56 @@ private static RobotDefinition loadSDFModel(String[] resourceDirectories, byte[]
    private static RobotDefinition loadURDFModel(String[] resourceDirectories, byte[] model, long modelHashCode, ClassLoader resourceClassLoader)
          throws JAXBException
    {
-      URDFModel urdfModel = URDFTools.loadURDFModel(new ByteArrayInputStream(model), Arrays.asList(resourceDirectories), resourceClassLoader);
+      Collection<InputStream> inputStreams = unpackModels(model);
+      URDFModel urdfModel = URDFTools.loadURDFModel(inputStreams, Arrays.asList(resourceDirectories), resourceClassLoader);
       RobotDefinition robotDefinition = URDFTools.toRobotDefinition(urdfModel);
       robotDefinition.setResourceClassLoader(resourceClassLoader);
       cachedImportedModels.put(modelHashCode, robotDefinition);
       return robotDefinition;
    }
 
+   /**
+    * Returns the InputStreams to their original format before being sent over the network
+    * @param models the combined byte array that holds all the InputStream's data
+    * @return the Collection of InputStream's in their original format
+    */
+   private static Collection<InputStream> unpackModels(byte[] models)
+   {
+      Collection<InputStream> inputStreams = new ArrayList<>();
+      long index = 0;
+
+      while(index < models.length)
+      {
+         // Reads the first 4 bytes to determine the length of the current InputStream
+         long length = ((long) (models[(int) index] & 0xFF) << 24) |
+                       ((long) (models[(int) index + 1] & 0xFF) << 16) |
+                       ((long) (models[(int) index + 2] & 0xFF) << 8) |
+                       (models[(int) index + 3] & 0xFF);
+
+         // Move index and read data for current
+         index += 4;
+
+         ByteArrayInputStream inputStream;
+
+         // The length is as expected which means we have metadata and should read a specific length
+         if (models.length >= length)
+         {
+            inputStream = new ByteArrayInputStream(models,(int) index, (int) length);
+         }
+         else
+         {  // This accounts for a model with no metadata, then we just read the stream without anything else
```

Comment:
This is how we preserve backward compatibility, for logs that don't have any metadata we just read the entire stream

Reply:
This works for new logs that have a composed URDF, those logs will have metadata. This works for URDF's that are in one file that also have metadata. And the last case is old logs that are one file and have no metadata, it works for that as well